src/os/uefi/console.rs

```diff
@@ -0,0 +1,66 @@
+use core::ffi::c_void;
+use core::fmt;
+use core::ptr::NonNull;
+use core::sync::atomic::{AtomicBool, Ordering};
+
+use one_shot_mutex::OneShotMutex;
+use uefi::table::boot::{EventType, Tpl};
+use uefi::table::{Boot, SystemTable};
+use uefi::Event;
+
+use crate::arch;
+
+pub enum Console {
+	None,
+	BootServices,
+	Native { console: arch::Console },
+}
+
+impl Console {
+	const fn new() -> Self {
+		Self::None
+	}
+
+	fn exit_boot_services(&mut self) {
```

Comment:
Do I get this correct: This is a callback for exiting the boot services? If so, I'd suggest modifying the name to reflect this. Also, I'm a huge fan of doc comments also for private fns :wink:

Reply:
Correct. I was inspired by https://github.com/rust-osdev/uefi-rs/blob/137932086eff0e0a7d0e5a63426a9e47f32439b3/uefi-services/src/lib.rs#L179.

For me, the few lines of code made the purpose self-explanatory, though we can always add docs, if people are confused.